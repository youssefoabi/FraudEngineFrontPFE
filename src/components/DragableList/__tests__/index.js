import { itRendersAllMutations } from '@cdiscount/jest-utils';

import DraggableList from '../';

const data = [
  {
    scopeId: 1,
    label: 'scope',
    order: 1,
  },
  {
    scopeId: 2,
    label: 'scope2',
    order: 2,
  },
];

const mutations = [
  {
    name: 'without props',
    props: {},
  },
  {
    name: 'with props',
    props: { data, renderItem: jest.fn, keyExtractor: jest.fn, onOrderChange: jest.fn },
  },
];

describe(DraggableList.name, () => {
  itRendersAllMutations(DraggableList, mutations);
});
