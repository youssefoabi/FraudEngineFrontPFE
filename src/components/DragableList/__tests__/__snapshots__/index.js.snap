// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableList DraggableList defaultProp functions runs defaultProps.keyExtractor() 1`] = `undefined`;

exports[`DraggableList DraggableList defaultProp functions runs defaultProps.onOrderChange() 1`] = `undefined`;

exports[`DraggableList DraggableList defaultProp functions runs defaultProps.renderItem() 1`] = `undefined`;

exports[`DraggableList DraggableList renders correctly without props 1`] = `""`;

exports[`DraggableList DraggableList renders correctly with props 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="row"
  item={true}
>
  <DragDropContext
    onDragEnd={[Function]}
  >
    <Connect(Droppable)
      direction="horizontal"
      droppableId={null}
      getContainerForClone={[Function]}
      ignoreContainerClipping={false}
      isCombineEnabled={false}
      isDropDisabled={false}
      mode="standard"
      renderClone={null}
      type="DEFAULT"
    >
      <Component />
    </Connect(Droppable)>
  </DragDropContext>
</WithStyles(ForwardRef(Grid))>
`;
